streams/streams_extra/run_stream

```diff
@@ -159,6 +159,20 @@ build_images()
  					continue
 				fi
  			fi
+			total_freemem=`free -k | grep Mem: | awk '{print $7}'`
+			free_mem=`echo \($total_freemem/100\)*90 | bc`
+			#
+			# We are using arrays of 8 byte elements, and there are 3 of them.
+			#
+			actual_mem_use=`echo $cache_size_kb*24 | bc`
+			#
+			# Do not build if the free_mem is less then the amount of memory
+			# this array size requires.
+			#
+			if [ $free_mem -lt $actual_mem_use ]; then
```

Comment:
I thought we were trying to standardize on `[[` instead of `[`, though I do see uses of both in this script which seems unfortunate.  If my recollection is accurate let's switch to that here but only handle the other cases with a separate issue / PR I think.